src/catalog/catalog_entry/table_catalog_entry.cpp

```diff
@@ -380,6 +390,91 @@ unique_ptr<CatalogEntry> TableCatalogEntry::SetDefault(ClientContext &context, S
 	return make_unique<TableCatalogEntry>(catalog, schema, (BoundCreateTableInfo *)bound_create_info.get(), storage);
 }
 
+unique_ptr<CatalogEntry> TableCatalogEntry::SetNotNull(ClientContext &context, SetNotNullInfo &info) {
+	
+	auto create_info = make_unique<CreateTableInfo>(schema->name, name);
+	// columns
+	vector<column_t> column_ids;
+	// types
+	vector<LogicalType> types;
+
+	for (idx_t i = 0; i < columns.size(); i++) {
+		auto copy = columns[i].Copy();
+		create_info->columns.push_back(move(copy));
+		// has to push all to avoid CheckZonemap() error
+		column_ids.push_back(i);
+		types.push_back(columns[i].type);
+	}
+
+	idx_t not_null_idx = GetColumnIndex(info.column_name);
+	bool has_not_null = false;
+	for (idx_t i = 0; i < constraints.size(); i++) {
+		auto constraint = constraints[i]->Copy();
+		if(constraint->type == ConstraintType::NOT_NULL){
+			auto &not_null = (NotNullConstraint&)*constraint;
+			if(not_null.index == not_null_idx){
+				has_not_null = true;
+			}
+		}
+		create_info->constraints.push_back(move(constraint));
+	}
+	// If this column has NULL value, throw exception
+	if (!has_not_null) {
+		//check!
+		// Transaction
+		auto &transaction = Transaction::GetTransaction(context);
+
+		// result
+		DataChunk result;
+		result.Initialize(types);
+
+		// filter
+		TableFilterSet table_filters;
+		table_filters.PushFilter(not_null_idx, make_unique<IsNullFilter>());
+
+		// state
+		TableScanState state;
+
+		storage->InitializeScan(transaction, state, column_ids, &table_filters);
+		storage->Scan(transaction, result, state, column_ids);
```

Comment:
This only scans a single column - this will not work for larger tables.

Reply:
Apologies I meant to say this only scans a single vector. This will scan at most 1024 values. `storage->Scan` needs to be called in a loop.

Reply:
Sorry, forget about my comment above, my test only works when append null value in the tail of the large table. You're right, I need to scan in a loop.